In [1]:
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline 
import seaborn as sns
import scipy.stats as stats

In [2]:
os.chdir('C:\\Users\\anush\\OneDrive\\Desktop\\Python Case Study\\4. Basic Statistics - Hypothesis Testing')

In [3]:
loans_data = pd.read_csv('LoansData.csv')
loans_data.head(3)

Amount.Requested  Amount.Funded.By.Investors Interest.Rate Loan.Length  \
0           20000.0                     20000.0         8.90%   36 months   
1           19200.0                     19200.0        12.12%   36 months   
2           35000.0                     35000.0        21.98%   60 months   

         Loan.Purpose Debt.To.Income.Ratio State Home.Ownership  \
0  debt_consolidation               14.90%    SC       MORTGAGE   
1  debt_consolidation               28.36%    TX       MORTGAGE   
2  debt_consolidation               23.81%    CA       MORTGAGE   

   Monthly.Income FICO.Range  Open.CREDIT.Lines  Revolving.CREDIT.Balance  \
0         6541.67    735-739               14.0                   14272.0   
1         4583.33    715-719               12.0                   11140.0   
2        11500.00    690-694               14.0                   21977.0   

   Inquiries.in.the.Last.6.Months Employment.Length  
0                             2.0          < 1 year  
1                             1.0           2 years  
2                             1.0           2 years

In [4]:
loans_data.dtypes

Amount.Requested                  float64
Amount.Funded.By.Investors        float64
Interest.Rate                      object
Loan.Length                        object
Loan.Purpose                       object
Debt.To.Income.Ratio               object
State                              object
Home.Ownership                     object
Monthly.Income                    float64
FICO.Range                         object
Open.CREDIT.Lines                 float64
Revolving.CREDIT.Balance          float64
Inquiries.in.the.Last.6.Months    float64
Employment.Length                  object
dtype: object

In [5]:
loans_data = loans_data.rename(columns={'Amount.Requested':'Amount_Requested','Amount.Funded.By.Investors':'Amount_Funded_By_Investors','Interest.Rate':'Interest_Rate','Loan.Length':'Loan_Length','Loan.Purpose':'Loan_Purpose','Debt.To.Income.Ratio':'Debt_To_Income_Ratio','Home.Ownership':'Home_Ownership','Monthly.Income':'Monthly_Income','FICO.Range':'FICO_Range','Open.CREDIT.Lines':'Open_CREDIT_Lines','Revolving.CREDIT.Balance':'Revolving_CREDIT_Balance','Inquiries.in.the.Last.6.Months':'Inquiries_in_the_Last_6_Months','Employment.Length':'Employment_Length'})

In [6]:
loans_data.head(3)

Amount_Requested  Amount_Funded_By_Investors Interest_Rate Loan_Length  \
0           20000.0                     20000.0         8.90%   36 months   
1           19200.0                     19200.0        12.12%   36 months   
2           35000.0                     35000.0        21.98%   60 months   

         Loan_Purpose Debt_To_Income_Ratio State Home_Ownership  \
0  debt_consolidation               14.90%    SC       MORTGAGE   
1  debt_consolidation               28.36%    TX       MORTGAGE   
2  debt_consolidation               23.81%    CA       MORTGAGE   

   Monthly_Income FICO_Range  Open_CREDIT_Lines  Revolving_CREDIT_Balance  \
0         6541.67    735-739               14.0                   14272.0   
1         4583.33    715-719               12.0                   11140.0   
2        11500.00    690-694               14.0                   21977.0   

   Inquiries_in_the_Last_6_Months Employment_Length  
0                             2.0          < 1 year  
1                             1.0           2 years  
2                             1.0           2 years

# BUSINESS PROBLEM: 
## Using lending club loans data, the team would like to test below hypothesis on how different factors effecing each other (Hint: You may leverage hypothesis testing using statistical tests) 

### a. Intrest rate is varied for different loan amounts (Less intrest charged for high loan amounts) 

Ho: There is no relationship between loan amount and interest rates
Ha: There is a relationship between loan amount and interest rates

CI = 95%
SL = 5%
pcritical = 0.05

Since we have to find relationship between numeric columns we will use correlation test.

In [7]:
loans_data['Interest_Rate'] = loans_data.Interest_Rate.str.replace('%','').astype(float)

In [8]:
loans_data[['Amount_Funded_By_Investors','Interest_Rate']].corr()

Amount_Funded_By_Investors  Interest_Rate
Amount_Funded_By_Investors                    1.000000       0.337454
Interest_Rate                                 0.337454       1.000000

There is a weak positive correlation between interest rate and amount funded by investors.

### b. Loan length is directly effecting intrest rate. 

Since one column is numeric and other categorical and we are comparing interest rates across 2 groups, we will be using independent t test.

Ho: There is no relationship between loan lenght and interest rate.
Ha: Loan lenght does not affect interest rate.

CI= 95%
SL= 5%
pcritical=0.05

In [9]:
pcritical=0.05
rate36 = loans_data[loans_data.Loan_Length == '36 months'].Interest_Rate.dropna()
rate60 = loans_data[loans_data.Loan_Length == '60 months'].Interest_Rate.dropna()
t_statistic,pvalue = stats.ttest_ind(rate36,rate60)

In [10]:
pvalue < pcritical

True

p is low null will go. Ha is accepted.
We are 95% confident that Loan lenght does not affect interest rates.

### c. Inrest rate varies for different purpose of loans 

#### Since, interest rate is numerical and loan purpose is categorical (with more than 2 groups) we will use anova test to check if interest rates differ across various loan purposes.

Ho: Different purpose of loans does not affect interest rates i.e. mean interest rates are same across all loan purposes.<br>
Ha: Purpose of loans does affect interest rates i.e. atleast one loan purpose has a different interest rate.

In [11]:
loans_data.Loan_Purpose.unique()

array(['debt_consolidation', 'credit_card', 'other', 'moving', 'car',
       'vacation', 'home_improvement', 'house', 'major_purchase',
       'educational', 'medical', 'wedding', 'small_business',
       'renewable_energy'], dtype=object)

In [12]:
#Samples

debt_cons_df = loans_data[loans_data.Loan_Purpose == 'debt_consolidation'].Interest_Rate.dropna()
credit_card_df =loans_data[loans_data.Loan_Purpose == 'credit_card'].Interest_Rate.dropna()
other_df = loans_data[loans_data.Loan_Purpose == 'other'].Interest_Rate.dropna()
moving_df = loans_data[loans_data.Loan_Purpose == 'moving'].Interest_Rate.dropna()
car_df = loans_data[loans_data.Loan_Purpose == 'car'].Interest_Rate.dropna()
vacation_df = loans_data[loans_data.Loan_Purpose == 'vacation'].Interest_Rate.dropna()
home_improvement_df = loans_data[loans_data.Loan_Purpose == 'home_improvement'].Interest_Rate.dropna()
house_df = loans_data[loans_data.Loan_Purpose == 'house'].Interest_Rate.dropna()
major_purchase_df = loans_data[loans_data.Loan_Purpose == 'major_purchase'].Interest_Rate.dropna()
educational_df = loans_data[loans_data.Loan_Purpose == 'educational'].Interest_Rate.dropna()
medical_df = loans_data[loans_data.Loan_Purpose == 'medical'].Interest_Rate.dropna()
wedding_df = loans_data[loans_data.Loan_Purpose == 'wedding'].Interest_Rate.dropna()
small_business_df = loans_data[loans_data.Loan_Purpose == 'small_business'].Interest_Rate.dropna()
renewable_energy_df = loans_data[loans_data.Loan_Purpose == 'renewable_energy'].Interest_Rate.dropna()

CI= 95%
SL= 5%

In [13]:
pcritical = 0.05

In [14]:
fstatistic,pvalue = stats.f_oneway(debt_cons_df,credit_card_df,other_df,moving_df,car_df,vacation_df,home_improvement_df,house_df,major_purchase_df,educational_df,medical_df,wedding_df,small_business_df,renewable_energy_df)

In [15]:
pcritical < pvalue

False

p is high, null will fly. Thus, Ho is accepted.
As per the sample used, we are 95% confident that different purpose of loans does not affect interest rates i.e. mean interest rates are same across all loan purposes.

### d. There is relationship between FICO scores and Home Ownership. It means that, People with owning home will have high FICO scores. 

In [16]:
# converting FICO ranges to numeric by calculating mid points

def fico_to_numeric(fico_range):
    if pd.isnull(fico_range):
        return None
    low,high= fico_range.split(sep='-')
    return (int(low)+int(high))/2

loans_data['FICO_numeric'] = loans_data['FICO_Range'].apply(fico_to_numeric)

In [17]:
loans_data = loans_data.dropna(subset='Home_Ownership')

#Hypothesis
Ho: There is no relationship between fico scores and home ownership i.e mean fico scores are same across all home ownership types.
Ha: There is a relationship between fico scores and home ownership i.e mean fico scores are different for atleast one home ownership type.

In [18]:
#Samples
type_mortgage = loans_data[loans_data.Home_Ownership == 'MORTGAGE'].FICO_numeric.dropna()
type_rent = loans_data[loans_data.Home_Ownership == 'RENT'].FICO_numeric.dropna()
type_own = loans_data[loans_data.Home_Ownership == 'OWN'].FICO_numeric.dropna()
type_other = loans_data[loans_data.Home_Ownership == 'OTHER'].FICO_numeric.dropna()
type_none = loans_data[loans_data.Home_Ownership == 'NONE'].FICO_numeric.dropna()

Since we have one numeric and one categorical variable having more than 2 categories, we will be using ANOVA test.

CI=95%
SL=5%

In [19]:
pcritical=0.05

In [20]:
fstatistic,pvalue = stats.f_oneway(type_mortgage,type_rent,type_own,type_other,type_none)

In [21]:
pcritical < pvalue

False

p is high, null will fly. Thus, Ho is accepted.
As per the sample used we can say that mean fico scores are same for all home ownership types

# BUSINESS PROBLEM - 2 
 
### BACKGROUND:  When an order is placed by a customer of a small manufacturing company, a price quote must be developed for that order. Because each order is unique, quotes must be established on an order-by-order basis by a pricing expert. The price quote process is laborintensive, as prices depend on many factors such as the part number, customer, geographic location, market, and order volume. The sales department manager is concerned that the pricing process is too complex, and that there might be too much variability in the quoted prices. An improvement team is tasked with studying and improving the pricing process. After interviewing experts to develop a better understanding of the current process, the team designed a study to determine if there is variability between pricing experts. That is, do different pricing experts provide different price quotes? Two randomly selected pricing experts, Mary and Barry, were asked to independently provide prices for twelve randomly selected orders. Each expert provided one price for each of the twelve orders. 
 
## BUSINESS PROBLEM: We would like to assess if there is any difference in the average price quotes provided by Mary and Barry.

In [22]:
price_quotes_df = pd.read_csv('Price_Quotes.csv')
price_quotes_df

Order_Number  Barry_Price  Mary_Price
0              1          126         114
1              2          110         118
2              3          138         114
3              4          142         111
4              5          146         129
5              6          136         119
6              7           94          97
7              8          103         104
8              9          140         127
9             10          152         133
10            11          108         103
11            12           97         108

Since both are giving independent quotations, we will be using independent t test

Ho: Barry_quotation = Mary_quotation
Ha: Barry_quotation != Mary_quotation

In [23]:
#Sample
Barry_quot = price_quotes_df.Barry_Price
Mary_quot = price_quotes_df.Mary_Price

CI = 95%
SL = 5%

In [24]:
pcritical = 0.05

In [25]:
tstatistic,pvalue = stats.ttest_ind(Barry_quot,Mary_quot)

In [26]:
pcritical < pvalue

True

p is low null will go. Thus, Ha is accepted.
As per the sample we can say that both the quotations provided by Barry and Mary are different.

# BUSINESS PROBLEM-3: 
### BACKGROUND: The New Life Residential Treatment Facility is a NGO that treats teenagers who have shown signs of mental illness. It provides housing and supervision of teenagers who are making the transition from psychiatric hospitals back into the community. Because many of the teenagers were severely abused as children and have been involved with the juvenile justice system, behavioral problems are common at New Life. Employee pay is low and staff turnover (attrition) is high. A reengineering program was instituted at New Life with the goals of lowering behavioral problems of the kids and decreasing employee turnover rates. As a part of this effort, the following changes were made:  Employee shifts were shortened from 10 hours to 8 hours each day.  Employees were motivated to become more involved in patient treatments. This included encouraging staff to run various therapeutic treatment sessions and allowing staff to have more say in program changes.  The activities budget was increased.  A facility-wide performance evaluation system was put into place that rewarded staff participation and innovation.  Management and staff instituted a program designed to raise expectations about appropriate behavior from the kids. This included strict compliance with reporting of behavioral violations, insistence on participation in therapeutic sessions, and a lowered tolerance for even moderate behavioral infractions. To determine the effectiveness of the reengineering effort, a data set comprised of pre- and post-reengineering periods was compiled. The information contains two measures of behavioral problems. A critical incident occurs when a resident goes AWOL (leaves the premises without permission), destroys property (e.g., punching a hole in a wall or throwing furniture through windows), is caught in possession of street drugs, or engages in assault against other residents or staff members. A teenager is temporarily removed from the facility when s/he is sent to jail or back to a psychiatric hospital 
 
## BUSINESS PROBLEM: 
### Determine what effect, if any, the reengineering effort had on the incidence behavioral problems and staff turnover. i.e To determine if the reengineering effort changed the critical incidence rate. Is there evidence that the critical incidence rate improved? 

In [27]:
treatment_df = pd.read_csv('Treatment_Facility.csv')
treatment_df.head(5)

Month Reengineer  Employee_Turnover       VAR4        VAR5
0      1      Prior             0.0000  24.390244   42.682927
1      2      Prior             6.0606  19.354839   25.806452
2      3      Prior            12.1212  35.087719  146.198830
3      4      Prior             3.3333  18.404908  110.429448
4      5      Prior            12.9032  17.964072   23.952096

In [28]:
#Hypothesis
Ho: There is no difference in avg rates between pre and post reengineering.
Ha: There is a difference in average rates between pre and post reengineering.

SyntaxError: invalid syntax (701034920.py, line 2)

In [ ]:
#Sample
prior_df = treatment_df[treatment_df.Reengineer == 'Prior']
post_df = treatment_df[treatment_df.Reengineer == 'Post']

In [ ]:
t_emp = stats.ttest_ind(prior_df.Employee_Turnover,post_df.Employee_Turnover)

In [ ]:
pcritical < t_emp.pvalue

p is low null will go. Ha accepted
#### As per the sample, There is a difference in average employee turnover rates between pre and post reengineering.

In [ ]:
t_VAR4 = stats.ttest_ind(prior_df.VAR4,post_df.VAR4)

In [ ]:
pcritical < t_VAR4.pvalue

p is high null will fly. Ho is accepted
#### As per the sample, There is no difference in avg VAR4 rates between pre and post reengineering.

In [ ]:
t_VAR5 = stats.ttest_ind(prior_df.VAR5,post_df.VAR5)

In [ ]:
pcritical < t_VAR5.pvalue

p is low null will go. Ha is accepted.
#### As per the sample, There is a difference in average VAR5 rates between pre and post reengineering.

# BUSINESS PROBLEM-4 
### BACKGROUND: Software development projects typically follow six basic phases: Requirements, design, implementation (and integration), testing (validation), deployment (installation) and maintenance. First, general requirements are gathered, and the scope of the functionality is defined. Then, alternative scenarios for the required functionality are developed and evaluated. Implementation, usually 50% or more of the development time, is the phase in which the design is translated into programs and integrated with other parts of the software – this is when software engineers actually develop the code. During the final phases, programs are tested, software is put into use, and faults or performance issues are addressed. ApDudes, a developer of applications for tablet computers, was having difficulty meeting project deadlines; only 10% of their projects had been completed within budget and on time last year and that was starting to hurt business. The group’s project manager was tasked with studying problems within the implementation phase. He found that software engineers were having difficulty prioritizing their work, and that they often became overwhelmed by the magnitude of the projects. As a result, two changes were made. Each project was broken down into smaller, distinct tasks, or jobs, and each job was assigned a priority. The project manager believes that this classification and prioritization system would speed the completion of high priority jobs, and thus lower overall project completion time 
## BUSINESS PROBLEM: 
### We will focus on the prioritization system. If the system is working, then high priority jobs, on average, should be completed more quickly than medium priority jobs, and medium priority jobs should be completed more quickly than low priority jobs. Use the data provided to determine whether this is, in fact, occurring.

In [ ]:
priority = pd.read_csv('Priority_Assessment.csv')
priority.head()

Ho: There is no difference between mean completion times between priorities
Ha: mean completion time differs for atleast one priority

In [ ]:
p_high = priority[priority.Priority == 'High'].Days
p_medium = priority[priority.Priority == 'Medium'].Days
p_low = priority[priority.Priority == 'Low'].Days

In [ ]:
CI = 95%
SL = 5%

In [ ]:
pcritical = 0.05

In [ ]:
fstatistic,pvalue = stats.f_oneway(p_high,p_medium,p_low)

In [ ]:
pcritical < pvalue

At 5% significance level we reject the null hypothesis as pcritical(0.05) < pvalue.(p is low, null will go. Ha is accepted.)
As per the sample, we can say that the mean completion time differs for high,medium,low priority jobs.

# BUSINESS PROBLEM-5 
### BackGround: Film on the Rocks is a summer movie series held at the world-renowned Red Rocks Amphitheatre, which is situated on a hillside of the Rocky Mountains in Morrison, Colorado fifteen miles west of Denver. The film series features classic films, and pre-show entertainment including bands and comedians. Among the features that have made Red Rocks an internationally-famous concert stage are sweeping views of Denver, outstanding acoustics, and cool, dry Colorado summers. The series is jointly promoted by the Denver Film Society (DFS) and the City and County of Denver’s Division of Theatres and Arenas (DT&A). It is marketed through various outlets including newspaper, radio, and the Red Rocks and Denver Film Society websites. Film on the Rocks patrons also benefit from corporate sponsorship. In return for on-site posters and banners at Red Rocks, and recognition in pre- show marketing materials, corporations donate funds that keep ticket prices low. Although the Red Rocks Amphitheatre provides a cinematic experience unlike any other venue, there are tradeoffs. Red Rocks is a farther commute for most people than the local movie theater or movie rental store. Given the uphill walk to the amphitheatre from the parking lot, getting there can be challenging. And, as an outdoor venue, the viewing experience is dependent on the weather. Patron satisfaction with Red Rocks as the venue for the film series is critical to its success. But, the series promoters would also like to increase attendance at the film series, and are unsure how to do this. Promoters recognize that they need a better understanding of the customer base, and of the current level of satisfaction. Knowing the demographics of those who attend the film series will help attract and expand corporate sponsorship. In addition, knowing which media outlets are most effective will provideinformation about how best to target future marketing campaigns. To this end, the promoters conducted surveys during a recent Film on the Rocks season. Questionnaires were handed out at the entrance. Volunteers walked through the crowd to remind people about the free soft drink given to those who returned the survey. 
 
## BUSINESS PROBLEM:
### Use the survey results to address the following questions 
###  What is the overall level of customer satisfaction? 

In [225]:
films_df = pd.read_csv('Films.csv')

In [226]:
films_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 330 entries, 0 to 329
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   _rowstate_      330 non-null    int64  
 1   Movie           330 non-null    object 
 2   Gender          330 non-null    object 
 3   Marital_Status  328 non-null    object 
 4   Sinage          328 non-null    float64
 5   Parking         328 non-null    float64
 6   Clean           327 non-null    float64
 7   Overall         328 non-null    float64
 8   Age             328 non-null    float64
 9   Income          314 non-null    float64
 10  Hear_About      323 non-null    object 
dtypes: float64(6), int64(1), object(4)
memory usage: 28.5+ KB


In [227]:
(films_df.isnull().sum()/len(films_df))*100

_rowstate_        0.000000
Movie             0.000000
Gender            0.000000
Marital_Status    0.606061
Sinage            0.606061
Parking           0.606061
Clean             0.909091
Overall           0.606061
Age               0.606061
Income            4.848485
Hear_About        2.121212
dtype: float64

In [228]:
films_df.dropna(inplace=True)

In [229]:
films_df.shape

(301, 11)

In [230]:
films_df['Gender']=np.where(films_df.Gender == '1','Male','Female')

In [231]:
films_df['Marital_Status']=np.where(films_df.Marital_Status == '1','Married','Single')

In [232]:
df_description = films_df.Overall.describe()
df_description

count    301.000000
mean       1.621262
std        0.675337
min        1.000000
25%        1.000000
50%        2.000000
75%        2.000000
max        5.000000
Name: Overall, dtype: float64

In [233]:
overall_mean = df_description['mean']
overall_std = df_description['std']
se = overall_std/(films_df.shape[0])**0.5

In [234]:
# Calculate 95% CI using standard error
lower_ci = overall_mean - 1.96 * se
upper_ci = overall_mean + 1.96 * se
print('95% confidence interval:')
print('Lower CI =',lower_ci)
print('Upper CI =',upper_ci)

95% confidence interval:
Lower CI = 1.5449679620169605
Upper CI = 1.697556954926561


We can say with 95% confidence that overall satisfaction of people ranges from 1.54 to 1.70

###  What factors are linked to satisfaction? 

In [235]:
#Anova/Overall satisfaction with all other variable---> when with categorical ind ttest/with numerical correlaion

#### Correlation for numerical columns(Sinage,Parking,Clean)

Ho: There is no relationship between overall and factors<br>
Ha: There is relatinship between overall and factors

In [236]:
numeric_columns = ['Sinage','Parking','Clean']
for col in numeric_columns:
    print('\n')
    print('Correlation between Overall and',col)
    print('\n')
    print(films_df[[col,'Overall']].corr())



Correlation between Overall and Sinage


           Sinage   Overall
Sinage   1.000000  0.410689
Overall  0.410689  1.000000


Correlation between Overall and Parking


          Parking   Overall
Parking  1.000000  0.547717
Overall  0.547717  1.000000


Correlation between Overall and Clean


            Clean   Overall
Clean    1.000000  0.385523
Overall  0.385523  1.000000


#### There is weak positive correlation between Clean and overall and moderate correlation between signage and parking

### Using Anovaand Ind T test for categorical variables

Using T test ind for Gender

In [237]:
male_data = films_df[films_df.Gender == 'Male']['Overall']
female_data = films_df[films_df.Gender == 'Female']['Overall']

In [238]:
tstatistic,pvalue = stats.ttest_ind(male_data,female_data)

In [239]:
#CI-95%
#SL-5%
pcritical= 0.05

In [240]:
pvalue > pcritical

True

Ho is accepted. There is no relationship between overall and factors.

###  What is the demographic profile of Film on the Rocks patrons? 

% of male female
% of married unmarried
age distribution
income distribution

In [241]:
films_df

_rowstate_                    Movie  Gender Marital_Status  Sinage  \
0             0  Ferris Buellers Day Off  Female         Single     2.0   
1             0  Ferris Buellers Day Off  Female         Single     1.0   
2             0  Ferris Buellers Day Off  Female         Single     2.0   
3             0  Ferris Buellers Day Off  Female         Single     1.0   
4             0  Ferris Buellers Day Off  Female         Single     1.0   
..          ...                      ...     ...            ...     ...   
325           0               Old School  Female         Single     1.0   
326           0               Old School    Male        Married     2.0   
327           0               Old School  Female        Married     2.0   
328           0               Old School    Male        Married     1.0   
329           0               Old School  Female         Single     2.0   

     Parking  Clean  Overall  Age  Income Hear_About  
0        2.0    2.0      2.0  3.0     1.0          5  
1        1.0    1.0      1.0  2.0     1.0          5  
2        4.0    3.0      2.0  4.0     1.0          5  
3        3.0    2.0      2.0  4.0     1.0          5  
4        1.0    1.0      1.0  3.0     3.0          1  
..       ...    ...      ...  ...     ...        ...  
325      2.0    1.0      1.0  2.0     1.0          1  
326      2.0    2.0      2.0  3.0     3.0          5  
327      1.0    1.0      2.0  2.0     2.0          5  
328      1.0    1.0      1.0  2.0     1.0          3  
329      2.0    3.0      1.0  2.0     1.0          5  

[301 rows x 11 columns]

In [243]:
demo_columns = ['Gender', 'Marital_Status', 'Age', 'Income', 'Hear_About']
demo_summary = {}
for cols in demo_columns:
    counts = films_df[cols].value_counts()
    percentages = ((films_df[cols].value_counts()/films_df[col].count())* 100).round(2)
    demo_summary[cols] = pd.DataFrame({'Counts':counts,'Percentages(%)':percentages})

for col,summary in demo_summary.items():
    print(col)
    print('\n')
    print(summary)
    print('\n')

Gender


        Counts  Percentages(%)
Gender                        
Female     208            69.1
Male        93            30.9


Marital_Status


                Counts  Percentages(%)
Marital_Status                        
Single             221           73.42
Married             80           26.58


Age


     Counts  Percentages(%)
Age                        
2.0     164           54.49
3.0     108           35.88
1.0      22            7.31
4.0       7            2.33


Income


        Counts  Percentages(%)
Income                        
1.0        135           44.85
3.0         86           28.57
2.0         80           26.58


Hear_About


            Counts  Percentages(%)
Hear_About                        
5              212           70.43
4               39           12.96
1               19            6.31
3               13            4.32
2               11            3.65
2,5              2            0.66
4,5              1            0.33
1,5              1  

###  In what media outlet(s) should the film series be advertised?

In [246]:
films_df.Hear_About.value_counts()

Hear_About
5      212
4       39
1       19
3       13
2       11
2,5      2
4,5      1
1,5      1
5,4      1
3,5      1
3,4      1
Name: count, dtype: int64

Since majority of people have heard from channel 5 i.e. word of mouth, we can increase referral programs and social outreach along with promotions on the website and TV.